src/control/cmd/daos_agent/fabric.go

```diff
@@ -273,6 +264,76 @@ func (n *NUMAFabric) setDefaultNUMANode() {
 	}
 }
 
+// Find finds a specific fabric device by name. There may be more than one domain associated.
+func (n *NUMAFabric) Find(name string) ([]*FabricInterface, error) {
+	if n == nil {
+		return nil, errors.New("nil NUMAFabric")
+	}
+
+	result := make([]*FabricInterface, 0)
+	for _, devs := range n.numaMap {
+		for _, fi := range devs {
+			if fi.Name == name {
+				result = append(result, copyFI(fi))
+			}
+		}
+	}
+
+	if len(result) > 0 {
+		return result, nil
+	}
+
+	return nil, fmt.Errorf("fabric interface %q not found", name)
+}
+
+// FindDevice looks up a fabric device with a given name, domain, and provider.
+// NB: The domain and provider are optional. All other parameters are required. If there is more
+// than one match, all of them are returned.
+func (n *NUMAFabric) FindDevice(name, domain, provider string) ([]*FabricInterface, error) {
```

Comment:
Instead of having multiple string parameters, some of which are optional, I wonder if it might be cleaner design to switch over more of these methods to using the `fiParams` struct to provide the equivalent of named parameters. It's more verbose at the call sites, I guess, but seems less error-prone.

Reply:
Agreed, I think that would be cleaner.